homeassistant/components/neato/vacuum.py

```diff
@@ -299,7 +293,8 @@ def extra_state_attributes(self) -> dict[str, Any]:
     @property
     def device_info(self) -> DeviceInfo:
         """Device info for neato robot."""
-        device_info = super().device_info
+        device_info = self._attr_device_info
+        assert device_info
```

Comment:
No change would be required, as you can see in the [`Entity` class](https://github.com/home-assistant/core/blob/0ae12ad08f4ffc6e1560665a44846784c79645d7/homeassistant/helpers/entity.py#L526)

In favor of not adding an `assert`, I would leave the code as it was.

Reply:
but `super.device_info()` is nullable, so that makes it harder :/

Reply:
Found a better solution. See the other conversation :)

```suggestion
        device_info = self._attr_device_info
```